keepercommander/subfolder.py

```diff
@@ -32,55 +32,83 @@ def find_folders(params, record_uid):
+
+    Get all the folders starting from the left end of component, plus the index of the first component that isn't present in the
+    folder cache.
+    """
+    remainder = 0
+    for index, component in enumerate(components):
+        temp_folder = contained_folder(params, folder, component)
+        if temp_folder is None:
+            break
+        folder = temp_folder
+        remainder = index + 1
+    return remainder, folder
+
+
+def is_abs_path(path_string):
+    """Return True iff path_string is an absolute path."""
+    return path_string.startswith('/') and not path_string.startswith('//')
+
+
+def path_split(params, folder, path_string):
+    """Split a path into directories with two replaces and a split."""
+    if is_abs_path(path_string):
+        folder = params.root_folder
+        path_string = path_string[1:]
+
+    components = [s.replace('\0', '/') for s in path_string.replace('//', '\0').split('/')]
+    return folder, [c.strip() for c in components]
+
+
 def try_resolve_path(params, path):
+    """
+    Look up the final keepercommander.subfolder.UserFolderNode and name of the final component(s).
+
+    If a record, the final component is the record.
+    If an existent folder, the final component is ''.
+    If a non-existent folder, the final component is the folders, joined with /, that do not (yet) exist..
+    """
     if type(path) is not str:
         path = ''
 
-    folder = params.folder_cache[params.current_folder] if params.current_folder in params.folder_cache else params.root_folder
-    if len(path) > 0:
-        if path[0] == '/':
-            folder = params.root_folder
-            path = path[1:]
-
-        start = 0
-        while True:
-            idx = path.find('/', start)
-            path_component = ''
-            if idx < 0:
-                if len(path) > 0:
-                    path_component = path.strip()
-            elif idx > 0 and path[idx - 1] == '\\':
-                start = idx + 1
-                continue
-            else:
-                path_component = path[:idx].strip()
-
-            if len(path_component) == 0:
-                break
-
-            folder_uid = ''
-            if path_component == '.':
-                folder_uid = folder.uid
-            elif path_component == '..':
-                folder_uid = folder.parent_uid or '/'
-            else:
-                for uid in folder.subfolders:
-                    sf = params.folder_cache[uid]
-                    if sf.name.strip().casefold() == path_component.casefold():
-                        folder_uid = uid
-                        break
-            if folder_uid:
-                folder = params.folder_cache[folder_uid] if folder_uid in params.folder_cache else params.root_folder
-            else:
-                break
-            if idx < 0:
-                path = ''
-                break
-
-            path = path[idx+1:]
-            start = 0
-
-    return folder, path
+    folder = (
```

Comment:
I think the following would be a bit simpler and you don't need the extra functions added to subfolder.py:
(import of posixpath is needed)
```
    no_slash_path = path.replace('//', '\0').strip()
    if no_slash_path.startswith('/'):
        abspath = no_slash_path[1:]
    elif params.current_folder == '':
        abspath = no_slash_path
    else:
        base_path = get_folder_path(params.current_folder)
        abspath = base_path + '/' + no_slash_path
    normpath = posixpath.normpath(abspath)

    if normpath.startswith('.'):
        return params.root_folder, ''
    else:
        subfoldernames = normpath.split('/')
        folder = params.root_folder
        for i, subfoldername in enumerate(subfoldernames):
            subfoldername = subfoldername.replace('\0', '/').strip()
            subfolder = [params.folder_cache[s] for s in folder.subfolders if params.folder_cache[s].name == subfoldername]
            if len(subfolder) == 0:
                break
            else:
                folder = subfolder[0]
        else:
            return folder, ''

        return folder, '/'.join(subfoldernames[i:]).replace('\0', '//')
```

Reply:
The separate functions enable thorough testing and complexity reduction.  The suggested alternative is far from simpler; take pity on the code's cyclomatic complexity: https://en.wikipedia.org/wiki/Cyclomatic_complexity

Reply:
Well break the above into functions. I still think it's simpler.